### ISSUE TITLE
Game crashed in editor after opening beatmap difficulty list

### ISSUE THREAD

**Issue by frukoyurdakul** [ppy/osu-stable-issues#331]
While I was testing something, I opened up a new difficulty, made my test, then while still on that test diff (I mean the diff was open in the editor), I removed the file from the game folder itself, then from `File --> Open Difficulty --> For Editing`  I opened the difficulty list. The deleted diff was not there, but when I clicked on the existing diff, game warned me about something like this:

`osu! encountered a problem and needs to be closed. Would you like to save?` with the options of `Yes, No, Cancel`. I clicked on `No` and it crashed.

Exception trace:

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at #=zfmcQ$iEIDw_9f5azsRft_EyeDnwU_WTetA==.#=z0X_l$GZGNpMU()
   at #=zfmcQ$iEIDw_9f5azsRft_EyeDnwU_WTetA==.#=zlKUnGybIlddS73wjaQ==(Object #=zp7GlPsw=, EventArgs #=z0c_uPy8=)
   at System.Windows.Forms.ListBox.OnSelectedIndexChanged(EventArgs e)
   at System.Windows.Forms.ListBox.WmReflectCommand(Message& m)
   at System.Windows.Forms.ListBox.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

```

I'm opening this issue because the game can warn us about it before actually crashing so I'm guessing it can be fixed. I don't know if this stack trace will be helpful, though, since it's a release build it's all obfuscated.

It should be reproduced easily with the steps above.

**Comment by smoogipoo** [comment]
You... went into the files and deleted the beatmap file directly? osu! even warns you to not do that if you modify the files manually before starting up osu!.

Don't do that. Out of scope.

**Comment by WinterIsPretty** [comment]
Lmao I did that once and I had to ask a mod to delete my map from osu xd
